analysis/metric_space.lean

```diff
@@ -480,6 +503,207 @@ instance metric_space_pi : metric_space (Πb, π b) :=
 
 end pi
 
+section second_countable
+
+/-- A separable metric space is second countable: one obtains a countable basis by taking
+the balls centered at points in a dense subset, and with rational radii. We do not register
+this as an instance, as there is already an instance going in the other direction
+from second countable spaces to separable spaces, and we want to avoid loops.-/
+lemma second_countable_of_separable_metric_space (α : Type u) [metric_space α] [separable_space α] :
+  second_countable_topology α :=
+let ⟨S, ⟨S_countable, S_dense⟩⟩ := separable_space.exists_countable_closure_eq_univ α in
+⟨⟨⋃x ∈ S, ⋃ (n : nat), {ball x (n⁻¹)},
+⟨show countable ⋃x ∈ S, ⋃ (n : nat), {ball x (n⁻¹)}, from
+begin
```

Comment:
`from begin` is not necessary.
remove the from. also in the next `show`, and maybe elsewhere.

Reply:
Done